### PR TITLE
Cerberus

### PR DESCRIPTION
**Pull request by rennerocha**

### CODE REVIEW

File: docs/source/item-validation.rst [hunk 1]
```diff
@@ -87,6 +87,10 @@ an example of a schema for the quotes item from the :doc:`tutorial </getting-sta
     ]
   }
 
+JSON Schema validation requires to include the location of your JSON schema file
```
Comment (rennerocha) [inline docs/source/item-validation.rst]: Improvements in documentation are always welcome! But considering that this branch is to implement Cerberus validation, this change you made is not related. A PR must not mix different subjects.

If you believe this line related to JSON Schema docs is important, the best way to suggest it to the project is creating a separate PR only with this change.

File: spidermon/contrib/scrapy/pipelines.py [hunk 2]
```diff
@@ -56,6 +56,7 @@ def set_validators(loader, schema):
         for loader, name in [
             (cls._load_jsonschema_validator, "SPIDERMON_VALIDATION_SCHEMAS"),
             (cls._load_schematics_validator, "SPIDERMON_VALIDATION_MODELS"),
+            (cls._load_cerberus_validator, "SPIDERMON_VALIDATION_CERBERUS")
```
Comment (rennerocha) [inline spidermon/contrib/scrapy/pipelines.py:58]: Related to my previous comment about the documentation, including a documentation about `SPIDERMON_VALIDATION_CERBERUS` is more suitable than the change you did related to JSON Schema.

File: spidermon/contrib/scrapy/pipelines.py [hunk 3]
```diff
@@ -96,6 +97,21 @@ def _load_jsonschema_validator(cls, schema):
             )
         return JSONSchemaValidator(schema)
 
+    @classmethod
+    def _load_cerberus_validator(cls, schema):
+        print("WORKING at loading")
```
Comment (rennerocha) [inline spidermon/contrib/scrapy/pipelines.py]: Including logs are good practice, but don't use `print` for logging. Use the [logging](https://docs.python.org/3/library/logging.html) library from Python. See [here](https://github.com/vipulgupta2048/spidermon/blob/master/spidermon/contrib/actions/slack/__init__.py#L7) an example of a better way to log.

This is a minor change and you can change it later, there is nothing wrong to add development logs (with your messages), but don't forget to remove them (or include useful info that should go to production).

File: spidermon/contrib/scrapy/pipelines.py [hunk 4]
```diff
@@ -96,6 +97,21 @@ def _load_jsonschema_validator(cls, schema):
...
+        if isinstance(schema, six.string_types):
+            schema = get_schema_from(schema)
```
Comment (rennerocha) [inline spidermon/contrib/scrapy/pipelines.py:102]: `get_schema_from` is a function that comes from `spidermon.contrib.validation.jsonschema.tools`. Maybe you will want to use it to get the schemas for Cerberus too and if so, keeping the function inside `jsonschema` is not a good idea (as it are not JSON schema only related). I would move it to `validation.utils` (see if the tests still pass).

File: spidermon/contrib/scrapy/pipelines.py [hunk 5]
```diff
@@ -96,6 +97,21 @@ def _load_jsonschema_validator(cls, schema):
...
+        if not isinstance(schema, dict):
+            raise NotConfigured(
+                "Invalid schema, jsonschemas must be defined as:\n"
```
Comment (rennerocha) [inline spidermon/contrib/scrapy/pipelines.py]: Fix the message. This is an error Cerberus related, not jsonschema.

File: spidermon/contrib/validation/cerberus/validator.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
+from __future__ import absolute_import
+
+from cerberus import Validator as something
```
Comment (rennerocha) [inline spidermon/contrib/validation/cerberus/validator.py]: You did right providing an alias for `cerberus.Validator` (as it conflicts with `spidermon.contrib.validation.validator.Validator`), but `something` is not a good alias as it has no useful meaning.

Some options would be:
`from cerberus import Validator as cerberus_validator`
or
`import cerberus` and use `cerberus.Validator` instead of `something`.

File: spidermon/contrib/validation/cerberus/validator.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+class CerberusValidator(Validator):
+    default_translator = None
+    name = "Cerberus"
+
+    def __init__(self, schema):
+        self._schema = schema
+
+    # Public method needed as Julio mentioned
+    def _validate(self, data):
```
Comment (rennerocha) [inline spidermon/contrib/validation/cerberus/validator.py]: Don't forget to include the `strict` argument here.

**Review by rennerocha — COMMENTED** [review]
I added some general comments over your code. Please let me know if you have any questions about it.

**Review by vipulgupta2048 — COMMENTED** [review]
Thanks, @rennerocha for the initial review. 
This push was spontaneous as it was reckless at the moment. But, I am happy that it happened. Atleast, some practices could be improved by the same. I have implemented most of the comments. Will push an update on the Cerberus pipeline soon enough.

File: docs/source/item-validation.rst (hunk 1 above)
Comment (vipulgupta2048) [reply docs/source/item-validation.rst]: This is a change that crept in as I rushed into committing all the changes. 
I was trying to solve the JSONschema issue at the time when I thought it's just a docs issue. 

Removed.

File: spidermon/contrib/scrapy/pipelines.py (hunk 2 above)
Comment (vipulgupta2048) [reply spidermon/contrib/scrapy/pipelines.py:58]: Roger that, will be implemented.

File: spidermon/contrib/scrapy/pipelines.py (hunk 3 above)
Comment (vipulgupta2048) [reply spidermon/contrib/scrapy/pipelines.py]: I have changed it now, I think its best to change these habits of debugging early. Good to know. 
Should have sanitized the code, will do it from my next commits.

File: spidermon/contrib/scrapy/pipelines.py [hunk 8]
```diff
@@ -3,19 +3,30 @@
...
 
-from spidermon.contrib.validation import SchematicsValidator, JSONSchemaValidator
-from spidermon.contrib.validation.jsonschema.tools import get_schema_from
+from spidermon.contrib.validation import SchematicsValidator, JSONSchemaValidator, CerberusValidator
+from spidermon.contrib.validation.utils import get_schema_from
 from schematics.models import Model
 
 from .stats import ValidationStatsManager
 
+logx = logging.getLogger(__name__)
```
Comment (ejulio) [inline spidermon/contrib/scrapy/pipelines.py]: Probably, these shouldn't be here

**Review by ejulio — CHANGES_REQUESTED** [review]
Missing tests for `CerberusValidator`.
It would be nice to have them in place before moving on

File: spidermon/contrib/scrapy/pipelines.py (hunk 8 above)
Comment (vipulgupta2048) [reply spidermon/contrib/scrapy/pipelines.py]: True that. Will remove it. 
Will add tests as well.

Comment (vipulgupta2048) [reply spidermon/contrib/scrapy/pipelines.py]: Thanks @ejulio

**Review by vipulgupta2048 — COMMENTED** [review]
Closing
